homeassistant/components/apsystems/entity.py

```diff
@@ -0,0 +1,44 @@
+"""APsystems base entity."""
+
+from __future__ import annotations
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ApSystemsData
+from .const import DOMAIN
+from .coordinator import ApSystemsDataCoordinator
+
+
+class ApSystemsEntity(Entity):
+    """Defines a base APsystems entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        data: ApSystemsData,
+    ) -> None:
+        """Initialize the APsystems entity."""
+        self._attr_unique_id = data.device_id
```

Comment:
No need to set the unique_id currently, as it is always overriden by the sensor entity.
It can be added later when/if it is needed by the other platforms.
```suggestion
```